File: bench/sieve.go
package main

import "fmt"

const limit = 10000000

func main() {
	composite := make([]bool, limit)
	count := 0
	for i := 2; i < limit; i++ {
		if !composite[i] {
			count++
			for j := i * i; j < limit; j += i {
				composite[j] = true
			}
		}
	}
	fmt.Println(count)
}

File: bench/sieve.rs
const LIMIT: usize = 100000000;
fn main() {
    let mut composite = vec![false; LIMIT];
    let mut count = 0usize;
    let mut i = 2;
    while i < LIMIT {
        if !composite[i] {
            count += 1;
            let mut j = i * i;
            while j < LIMIT { composite[j] = true; j += i; }
        }
        i += 1;
    }
    println!("{}", count);
}
